frontend/src/hooks/tests/useToolbarButtons.spec.tsx

```diff
@@ -0,0 +1,187 @@
+    Object.defineProperty(overflowAndExitRef.current, 'clientWidth', {
+      configurable: true,
+      writable: true,
+      value: 108,
+    });
+    vi.spyOn(window, 'getComputedStyle').mockImplementation((elt: Element) => {
+      if (elt.id === 'toolbar-ref') {
+        return {
+          paddingLeft: '30px',
+          paddingRight: '30px',
+        } as CSSStyleDeclaration;
+      }
+      if (elt.id === 'time-room-name') {
+        return {
+          marginRight: '12px',
+        } as CSSStyleDeclaration;
+      }
+      return {
+        marginLeft: '12px',
+      } as CSSStyleDeclaration;
+    });
+  });
+
+  afterEach(() => {
+    vi.restoreAllMocks();
+  });
+
+  afterAll(() => {
+    vi.unstubAllGlobals();
+  });
+
+  it('returns nothing when there is no space in the toolbar', async () => {
+    Object.defineProperty(toolbarRef.current, 'clientWidth', {
+      configurable: true,
+      writable: true,
+      value: 412,
+    });
+
+    const { result } = renderHook(() =>
+      useToolbarButtons({
+        toolbarRef,
+        mediaControlsRef,
+        overflowAndExitRef,
+        rightPanelControlsRef,
+        numberOfToolbarButtons,
+        timeRoomNameRef,
+      })
+    );
+
+    const { centerButtonLimit, rightButtonLimit, displayTimeRoomName } = result.current;
+    expect(centerButtonLimit).toBe(0);
+    expect(rightButtonLimit).toBe(0);
+    expect(displayTimeRoomName).toBe(false);
+  });
+
+  it('returns only the first button when there is space for one button in the toolbar', () => {
+    Object.defineProperty(toolbarRef.current, 'clientWidth', {
+      configurable: true,
+      writable: true,
+      value: 466,
+    });
+
+    const { result } = renderHook(() =>
+      useToolbarButtons({
+        toolbarRef,
+        mediaControlsRef,
+        overflowAndExitRef,
+        rightPanelControlsRef,
+        numberOfToolbarButtons,
+        timeRoomNameRef,
+      })
+    );
+
+    const { centerButtonLimit, rightButtonLimit, displayTimeRoomName } = result.current;
+
+    expect(centerButtonLimit).toBe(1);
+    expect(rightButtonLimit).toBe(1);
+    expect(displayTimeRoomName).toBe(false);
+  });
+
+  it('returns only the first three button when there is space for three button in the toolbar', () => {
+    Object.defineProperty(toolbarRef.current, 'clientWidth', {
+      configurable: true,
+      writable: true,
+      value: 586,
+    });
+
+    const { result } = renderHook(() =>
+      useToolbarButtons({
+        toolbarRef,
+        mediaControlsRef,
+        overflowAndExitRef,
+        rightPanelControlsRef,
+        numberOfToolbarButtons,
+        timeRoomNameRef,
+      })
+    );
+
+    const { centerButtonLimit, rightButtonLimit, displayTimeRoomName } = result.current;
+
+    expect(centerButtonLimit).toBe(3);
+    expect(rightButtonLimit).toBe(3);
+    expect(displayTimeRoomName).toBe(false);
+  });
+
+  it('will display the TimeRoomName component and the max amount of right panel buttons', () => {
```

Comment:
nit: this test name is a little too much to the point, maybe we can do something that is more similar to other test names in this file such as 
`displays the time and room name as well as all the right panel buttons` 
or something like that 🙏